detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/MissingUseCall.kt

```
@@ -0,0 +1,134 @@
+import io.gitlab.arturbosch.detekt.api.Debt
+import io.gitlab.arturbosch.detekt.api.Entity
+import io.gitlab.arturbosch.detekt.api.Issue
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.api.Severity
+import io.gitlab.arturbosch.detekt.api.internal.RequiresTypeResolution
+import io.gitlab.arturbosch.detekt.rules.fqNameOrNull
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.js.translate.callTranslator.getReturnType
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtObjectLiteralExpression
+import org.jetbrains.kotlin.resolve.calls.util.getResolvedCall
+import org.jetbrains.kotlin.resolve.calls.util.getType
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameOrNull
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.types.typeUtil.supertypes
+
+/**
+ * Prefer using `use` function with `Closeable` or `AutoCloseable`. As `use` function closes it resource correctly
+ * whether an exception is thrown or not.
+ *
+ * <noncompliant>
+ * val myCloseable = MyCloseable()
+ * // do stuff with myCloseable
+ *
+ * MyClosable().doStuff()
+ *
+ * functionThatReturnsClosable().doStuff()
+ *
+ * </noncompliant>
+ *
+ * <compliant>
+ * MyCloseable().use {
+ *     // do stuff with myCloseable
+ * }
+ *
+ * MyClosable().use { it.doStuff() }
+ *
+ * functionThatReturnsClosable().use { it.doStuff() }
+ * </compliant>
+ */
+@RequiresTypeResolution
+class MissingUseCall(config: Config = Config.empty) : Rule(config) {
+    override val issue: Issue = Issue(
+        javaClass.simpleName,
+        Severity.Warning,
+        "Usage of `Closeable` detected with `use` call. Using `Closeable` without `use` can be problematic " +
+            "as closing `Closeable` may throw exception.",
```

Also I'm unsure this sentence is entirely correct, as the point is about users forgetting to close `Closeable`.

---

`as the point is about users forgetting to close Closeable.` So that's also an issue I think your mentioned issue slightly bigger one. 

But even if you close the `Closeable`, `use` also does more like properly closing it in `finally` block also if you have some exception while using `Closebale` it records that and then in the `finally` block if we get another exception then add that recorded exception as suppressed throwable of the new exception

It also properly checks if Kotlin supports suppressed exceptions or not. If it doesn't then the original exception gets ignored 😅

---

Although I have updated the doc